src/TransactionFetcher.ts

```diff
@@ -27,15 +33,21 @@ export default class TransactionFetcher {
     const entity = (Object.keys(data)[0]);
     const entityRecord = data[entity][0];
     const query = EntityGraphQueries[entity];
-    // Fetch uts based on entity & contract address from ContractEntity model and update the
-    // variables object uts field. <PLACEHOLDER>
+
+    const contractEntityRecord = await this.contractEntityRepository.get(
+      entityRecord.contractAddress,
+      entity,
+    );
+
+    const uts = contractEntityRecord ? contractEntityRecord.timestamp : 0;
```

Comment:
I would rather check against null, and again, the case 0 is possible?

Reply:
Yea, make sense. I am checking and throwing error.